### ISSUE TITLE
Copilot Stops Working in Saved Quarto (.qmd) Files in RStudio

### ISSUE THREAD

**Issue by mrembert** [rstudio/rstudio#15539]
### System details

    RStudio Edition : Desktop
    RStudio Version :  2024.12.0 
    OS Version      :  Windows 11 Enterprise Version 21H2
    R Version       :  4.3.1

### Steps to reproduce the problem

1. Open RStudio.
2. Create a new Quarto document (File -> New File -> Quarto Document).
3. Start typing R code in the document. GitHub Copilot suggestions work as expected.
4. Save the Quarto file (e.g., as test.qmd).
5. Continue typing R code. Copilot suggestions no longer appear.
6. Manually trigger a Copilot completion request using Ctrl+\. No Copilot activity is observed in the RStudio console or the Copilot logs.

### Describe the problem in detail
Detailed Description:

Within the RStudio IDE, GitHub Copilot exhibits inconsistent behavior when working with Quarto (.qmd) files. While Copilot functions as expected in unsaved Quarto files and both saved and unsaved R files (.r), it completely ceases to provide code suggestions or completions in Quarto files once they are saved to disk.

Specifically, the following behaviors are observed:

- **Interruption of Service:** After saving a Quarto file, Copilot stops providing automatic code suggestions as you type.
- **Unresponsive to Manual Requests:** Manual requests for Copilot completions (triggered by the keyboard shortcut Ctrl+\) are also ignored. There is no visible activity or response from Copilot in the RStudio console or the Copilot logs when a manual request is made within a saved Quarto file.
- **Inconsistent with File Type:** This behavior is specific to saved Quarto files. Copilot continues to function correctly in unsaved Quarto files and in both saved and unsaved R files.
- **Independent of Project Context:** The issue occurs regardless of whether the Quarto file is part of an RStudio project or a standalone file.
- **Copilot Logs:** The Copilot logs indicate that the extension is successfully indexing all project files, including saved Quarto files. However, no log entries are generated when a completion request is made (either automatically or manually) from within a saved Quarto file, suggesting that RStudio is not forwarding these requests to the Copilot extension.

### Describe the behavior you expected
GitHub Copilot should continue to provide code suggestions and completions in saved Quarto files, just as it does in unsaved Quarto files and in R files.

- [x ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@mrembert Thank you for raising this! And excellent bug report.

I can reproduce this in RStudio Desktop 2024.12.0+467 on Windows 11. Problem is both in code and text portions. Everything's fine with R and R Markdown. 

@kevinushey Copilot's behavior is the same as with #15507, although now with saved Quarto files instead of unsaved files.

**Comment by CorneeldH** [comment]
Same issue on Mac, I'm happy to test potential fixes.

**Comment by kevinushey** [comment]
Thanks everyone -- we have a potential fix in the latest [daily builds](https://dailies.rstudio.com/rstudio/kousa-dogwood/) available at https://dailies.rstudio.com/rstudio/kousa-dogwood/, if you'd like to give it a test. Please download 2024.12.1+519 or newer.

**Comment by CorneeldH** [comment]
Great, I just tested +520 on Mac I have auto-complete in existing Quarto again (I haven't tested anything else). Thanks for the quick fix!

**Comment by danielvartan** [comment]
I'm experiencing the same issue on Linux. I will test the suggested fix and report back with the results.

```
RStudio Edition  : Desktop
RStudio Version :  2024.12.0 Build 467
OS Version          :  Ubuntu 24.04.1 LTS
R Version             :  4.4.2
```

**Comment by ronblum** [comment]
Verified fixed in RStudio Desktop 2024.12.1+522 on Windows 11. Checked

- R, R Markdown, and Quarto
- With new file, newly-saved file, and reopened file
- For Rmd and Qmd, checked
    - Inside a code chunk
    - In body text

**Comment by fgardavaud** [comment]
@mrembert  thanks a lot for this issue oponing and @kevinushey for the solution provided.
